extended_tasks/mt_bench/judges.py

```diff
@@ -0,0 +1,136 @@
+# MIT License
+
+# Copyright (c) 2024 The HuggingFace Team
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+import ast
+import json
+import re
+import time
+from abc import ABC
+from typing import Optional
+
+from openai import OpenAI
+
+from lighteval.logging.hierarchical_logger import hlog_warn
+
+
+# Abstract class for a judge
+class Judge(ABC):
+    def evaluate_answer(answers, questions, references) -> tuple[str, list[dict[str, str]], str]:
+        pass
+
+
+class JudgeOpenAI(Judge):
+    def __init__(self, model: str, seed: int, temperature: float, templates_path: str):
+        self.client = OpenAI()
```

Comment:
Does it need a token? If it extracts it from the env, maybe we should log a message asking the user to log itself first